frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -133,7 +133,8 @@ export async function request(
   path: string,
   params: RequestParams = {},
   autoLogoutOnAuthError: boolean = true,
-  useCluster: boolean = true
+  useCluster: boolean = true,
+  queryParams?: QueryParameters
```

Comment:
I wonder if we should deprecated the current version of request and use all these options as part of the `RequestParams` so it scales better.

Reply:
If my understanding is correct,In the `request` implementation `RequestParams` is used to pass `RequestHeaders` and the newly added `QueryParams` cannot be merged together with that.

Reply:
We should rename the `params` as `headers` then. This function's parameters are also getting very long, so we should eventually create a `request(path: string, params: RequestParams);` where request params are all these parameters supplied as an object (as it will scale better for the future).

This means also deprecating the current way.